stdlib/public/core/ArrayBuffer.swift

```diff
@@ -247,29 +247,26 @@ extension _ArrayBuffer {
       // Look for contiguous storage in the NSArray
       let nonNative = self._nonNative
       let cocoa = _CocoaArrayWrapper(nonNative)
-      let cocoaStorageBaseAddress = cocoa.contiguousStorage(self.indices)
-
-      if cocoaStorageBaseAddress != nil {
-        return _SliceBuffer(
-          owner: nonNative,
-          subscriptBaseAddress: UnsafeMutablePointer(cocoaStorageBaseAddress),
-          indices: subRange,
-          hasNativeBuffer: false)
+      guard let cocoaStorageBaseAddress = cocoa.contiguousStorage(self.indices) else {
```

Comment:
I prefer this the old way.  `guard` is supposed to be a quick-exit path, and in this case, `if` plays that role well.